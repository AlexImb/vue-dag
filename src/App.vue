<template>
  <main>
    <vue-dag v-model="graphData" />
  </main>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';
import VueDag from './VueDag.vue';
import { GraphData } from 'types';

@Component({
  components: { VueDag },
})
export default class App extends Vue {
  graphData: GraphData = {
    config: {
      scale: 1,
    },
    nodes: [
      { id: 0, content: 'content 0', x: 500 * Math.random(), y: 500 * Math.random() },
      { id: 1, content: 'content 1', x: 500 * Math.random(), y: 500 * Math.random() },
      { id: 2, content: 'content 2', x: 500 * Math.random(), y: 500 * Math.random() },
    ],
    edges: [{ id: 0, content: 'link 0', from: 0, to: 1 }],
  };
}
</script>

<style lang="scss" scoped>
main {
  height: 100vh;
  width: 100vw;
  overflow: auto;
}
</style>
