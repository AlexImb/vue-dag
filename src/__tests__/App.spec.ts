import { shallowMount, Wrapper } from '@vue/test-utils';
import App from '@/App.vue';

let wrapper: Wrapper<App>;

describe('App', () => {
  beforeEach(() => {
    wrapper = shallowMount(App);
  });

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
