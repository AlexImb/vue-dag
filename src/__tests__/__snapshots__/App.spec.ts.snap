// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<main>
  <section class="section">
    <h1 class="title">Vue DAG Demo</h1>
    <div class="columns">
      <div class="column is-4"></div>
      <div class="column">
        <vue-dag-stub value="[object Object]"></vue-dag-stub>
      </div>
    </div>
  </section>
</main>
`;
